submanifests/ci.yaml

```diff
@@ -0,0 +1,26 @@
+manifest:
+  remotes:
+    - name: danieldegrasse
+      url-base: https://github.com/danieldegrasse
+    - name: tenstorrent
+      url-base: https://github.com/tenstorrent
+  defaults:
+    remote: tenstorrent
+  group-filter:
+    - -ci
+  # zephyr-keep-sorted-start re(^\s+\- name:)
+  projects:
+    - name: openocd
+      revision: 720c2b6023433ae4d11d929a74c71e96f5d8159f
+      path: ci/openocd
+      remote: danieldegrasse
+      submodules: true
+      groups:
+        - ci
+    - name: tt-metal
+      revision: v0.56.0
```

Comment:
+1

This is good. It will allow us to update more easily without breaking anything. Can we simplify cloning?

https://github.com/zephyrproject-rtos/example-application/blob/main/.github/workflows/build.yml#L27

Metal will need to be able to test for certain features before enabling them though, since they do not have a similar means of tracking changes.

Reply:
The method I want to use here is sequencing like this:
- metal makes a PR on their side to enable a feature
- We create a PR bumping the metal revision, and turning on whatever feature they want to test
- Our CI runs metal, and validates that tests pass with the new feature enabled